src/edu/pugetsound/mathcs/architecture_nlp/features/spf/GenerateInteractor.java

```diff
@@ -0,0 +1,297 @@
+import edu.cornell.cs.nlp.spf.parser.ccg.cky.CKYBinaryParsingRule;
+import edu.cornell.cs.nlp.spf.parser.ccg.cky.CKYUnaryParsingRule;
+import edu.cornell.cs.nlp.spf.parser.ccg.cky.logger.ChartLogger;
+import edu.cornell.cs.nlp.spf.parser.ccg.cky.single.CKYParser;
+import edu.cornell.cs.nlp.spf.parser.ccg.cky.sloppy.BackwardSkippingRule;
+import edu.cornell.cs.nlp.spf.parser.ccg.cky.sloppy.ForwardSkippingRule;
+import edu.cornell.cs.nlp.spf.parser.ccg.cky.sloppy.SimpleWordSkippingLexicalGenerator;
+import edu.cornell.cs.nlp.spf.parser.ccg.factoredlex.features.FactoredLexicalFeatureSet;
+import edu.cornell.cs.nlp.spf.parser.ccg.features.basic.DynamicWordSkippingFeatures;
+import edu.cornell.cs.nlp.spf.parser.ccg.features.basic.LexicalFeaturesInit;
+import edu.cornell.cs.nlp.spf.parser.ccg.features.basic.scorer.ExpLengthLexicalEntryScorer;
+import edu.cornell.cs.nlp.spf.parser.ccg.features.lambda.LogicalExpressionCoordinationFeatureSet;
+import edu.cornell.cs.nlp.spf.parser.ccg.lambda.pruning.SupervisedFilterFactory;
+import edu.cornell.cs.nlp.spf.parser.ccg.model.LexiconModelInit;
+import edu.cornell.cs.nlp.spf.parser.ccg.model.Model;
+import edu.cornell.cs.nlp.spf.parser.ccg.model.ModelLogger;
+import edu.cornell.cs.nlp.spf.parser.ccg.normalform.NormalFormValidator;
+import edu.cornell.cs.nlp.spf.parser.ccg.normalform.hb.HBComposedConstraint;
+import edu.cornell.cs.nlp.spf.parser.ccg.rules.lambda.PluralExistentialTypeShifting;
+import edu.cornell.cs.nlp.spf.parser.ccg.rules.lambda.ThatlessRelative;
+import edu.cornell.cs.nlp.spf.parser.ccg.rules.lambda.typeraising.ForwardTypeRaisedComposition;
+import edu.cornell.cs.nlp.spf.parser.ccg.rules.lambda.typeshifting.PrepositionTypeShifting;
+import edu.cornell.cs.nlp.spf.parser.ccg.rules.primitivebinary.application.BackwardApplication;
+import edu.cornell.cs.nlp.spf.parser.ccg.rules.primitivebinary.application.ForwardApplication;
+import edu.cornell.cs.nlp.spf.parser.ccg.rules.primitivebinary.composition.BackwardComposition;
+import edu.cornell.cs.nlp.spf.parser.ccg.rules.primitivebinary.composition.ForwardComposition;
+import edu.cornell.cs.nlp.spf.parser.graph.IGraphParser;
+import edu.cornell.cs.nlp.spf.test.Tester;
+import edu.cornell.cs.nlp.spf.test.stats.ExactMatchTestingStatistics;
+import edu.cornell.cs.nlp.utils.collections.SetUtils;
+import edu.cornell.cs.nlp.utils.function.PredicateUtils;
+import edu.cornell.cs.nlp.utils.log.ILogger;
+import edu.cornell.cs.nlp.utils.log.Log;
+import edu.cornell.cs.nlp.utils.log.LogLevel;
+import edu.cornell.cs.nlp.utils.log.Logger;
+import edu.cornell.cs.nlp.utils.log.LoggerFactory;
+import edu.pugetsound.mathcs.architecture_nlp.features.spf.Interactor;
+/**
+ * Cross validation experiment for GeoQuery using fold0 for testing. This class
+ * is intended to illustrate how an experiment is structured. For complete
+ * experiments see the accompanying ExPlat files.
+ *
+ * @author Yoav Artzi & Jared Polonitza
```

Comment:
Underneath here, add a comment explaining (1) what a GenerateInteractor is and what it does and (2) how this file differs from its counterpart in the SPF jar file